weatherman.py

```diff
@@ -3,68 +3,141 @@
+    max_temp_date = None
+    min_temp_date = None
+    humidity_date = None
 
-global_max_temp = float('-inf')
-global_min_temp = float('inf')
-highest_humidity = float('-inf')
-humidity_count = 0
-humidity_percentage = 0
-max_temp_date = ""
-min_temp_date = ""
-humidity_date = ""
+    for line in my_weatherlist:
+        try:
+            temperature = float(line[1])  
+            humidity = float(line[-1])
+            date_str = line[0]
+            date_obj = datetime.datetime.strptime(date_str, "%Y-%m-%d")
 
+            if temperature > global_max_temp:
+                global_max_temp = temperature
+                max_temp_date =  date_obj
+            if temperature < global_min_temp:
+                global_min_temp = temperature
+                min_temp_date = date_obj
+            if humidity > highest_humidity:
+                highest_humidity = humidity 
+                humidity_date = date_obj
 
-for file_path in files:
-    if os.path.exists(file_path):
-        try:
-            read_files(file_path)
-        except Exception as e:
-            file_path: {e}
-
-for line in my_weatherlist:
-    try:
-        temperature = float(line[1])  
-        humidity = float(line[-1])
-        date_str = line[0]
-        date_obj = datetime.datetime.strptime(date_str, "%Y-%m-%d")
-
-        if temperature > global_max_temp:
-           global_max_temp = temperature
-           max_temp_date = date_obj
-        if temperature < global_min_temp:
-           global_min_temp = temperature
-           min_temp_date = date_obj
-        if humidity > highest_humidity:
-           highest_humidity = humidity 
-           humidity_date = date_obj
-
-        humidity_percentage += humidity
-        humidity_count += 1
-        humidity_percentage = (humidity/highest_humidity)* 100.0
-
-
-    except (ValueError, IndexError) as e:
-        line: {e}
-        continue
-
-print(f'Highest: {global_max_temp} on {max_temp_date.strftime("%B %d")}')
-print(f'Lowest: {global_min_temp} on {min_temp_date.strftime("%B %d")}')
-print(f'Humidity: {humidity_percentage} on {humidity_date.strftime("%B %d")}')
+            humidity_percentage += humidity
+            humidity_count += 1
+            humidity_percentage = (humidity/highest_humidity)* 100.0
+
+        except (ValueError, IndexError) as e:
+            continue
+    if max_temp_date:
+        max_temp_str = max_temp_date.strftime("%B %d")
+    else:
+        max_temp_str = "N/A"
+
+    if min_temp_date:
+        min_temp_str = min_temp_date.strftime("%B %d")
+    else:
+        min_temp_str = "N/A"
+
+    if humidity_date:
+        humidity_date_str = humidity_date.strftime("%B %d")
+    else:
+        humidity_date_str = "N/A"
+
+    print(f'Highest: {global_max_temp} on {max_temp_str}')
+    print(f'Lowest: {global_min_temp} on {min_temp_str}')
+    print(f'Humidity: {humidity_percentage} on {humidity_date_str}')
+
+def handling_e_command(year_month):
+    years, months = map_months(year_month)
```

Comment:
As you have created different methods for both of the flags and for `-e` flag you don't need to map months so you should remove map_months method from here.